Source/Atomic/UI/UIFinderWindow.h

```diff
@@ -0,0 +1,86 @@
+//
+// Copyright (c) 2014-2017, THUNDERBEAST GAMES LLC All rights reserved
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+#include <TurboBadger/tb_atomic_widgets.h>
+
+#include "UIWindow.h"
+
+namespace Atomic
+{
+
+class UIPromptWindow;
+
+class UIFinderWindow : public UIWindow
+{
+    ATOMIC_OBJECT(UIFinderWindow, UIWindow)
+
+public:
+
+    UIFinderWindow(Context* context, UIWidget* target, const String& id, bool createWidget = true);
+    virtual ~UIFinderWindow();
+
+    void FindFile(const String& title, const String& preset, int dimmer=0, int width=0, int height=0);
+    void FindPath(const String& title, const String& preset, int dimmer=0, int width=0, int height=0);
+
+protected:
+
+    void HandleCreateBookmark(StringHash eventType, VariantMap& eventData); /// event handler for bookmark creation
+    void HandleCreateFolder(StringHash eventType, VariantMap& eventData); /// event handler for folder creation
+    virtual bool OnEvent(const tb::TBWidgetEvent &ev); /// general event handler
+    UIWidget* GetWindowWidget(); /// get internal widgets
+    UIWidget* GetPathWidget();
+    UIWidget* GetResultWidget();
+    UIWidget* GetBookmarksWidget();
+    UIWidget* GetFilesWidget();
+   
+private:
+
+    void PresetCurrentPath( const String& preset );  /// location where the finder starts
+    void SetCurrentPath( const String& string );  /// set the current path value
+    void ComposePath (const String& string );  /// using the list, jam things together, we'll either get another path or a file.
+    void CreateBookmarks(); /// create the list of bookmarks, per platform
+    void GoFolderUp(); /// go up one folder
+    void UpdateUiPath ();  /// move current path to widget
```

Comment:
Are the names here ```Ui``` instead of ```UI``` due to script generation, if so we should fix the binding gen for these instead of locking an inconsistent API string case